packages/shared/lib/ledger.ts

```diff
@@ -6,62 +6,73 @@ import { api } from 'shared/lib/wallet'
 import { get, writable } from 'svelte/store'
 import { localize } from './i18n'
 
-export const ledgerSimulator = true
+export const ledgerSimulator = false
 export const isLedgerConnected = writable<boolean>(true)
 export const isLedgerLegacyConnected = writable<boolean>(true)
 
-const DEFAULT_LEDGER_STATUS_POLL_INTERVAL = 1000
+const DEFAULT_LEDGER_STATUS_POLL_INTERVAL = 1500
 
-export function getLedgerDeviceStatus(): void {
-    api.getLedgerDeviceStatus(ledgerSimulator, {
+let polling = false
+let ledgerPollInterval
+
+export function getLedgerDeviceStatus(onConnected, onCancel) {
+    const cancel = () => {
+        stopPollingLedgerStatus()
+        onCancel()
+    }
+    const pollLedgerStatus = () => api.getLedgerDeviceStatus(ledgerSimulator, {
         onSuccess(response) {
             let _isLedgerConnected = response.payload?.type === LedgerStatus.Connected
             if (_isLedgerConnected) {
-                closePopup()
+                stopPollingLedgerStatus()
+                if (get(popupState).active) {
```

Comment:
Opt: You could change this into the following:
`get(popupState).active && closePopup()`